tfhe/src/integer/bigint/algorithms.rs

```diff
@@ -182,25 +223,89 @@ where
     (quotient, remainder)
 }
 
+pub(crate) fn slow_div_signed<T>(numerator: T, divisor: T) -> (T, T)
+where
+    T: Numeric
+        + ShlAssign<u32>
+        + Shl<u32, Output = T>
+        + Shr<u32, Output = T>
+        + BitOrAssign<T>
+        + SubAssign<T>
+        + BitAnd<T, Output = T>
+        + Ord
+        + Neg<Output = T>
+        + Add<T, Output = T>
+        + Sub<T, Output = T>,
+{
+    assert!(divisor != T::ZERO);
+
+    let positive_numerator = if numerator < T::ZERO {
+        -numerator
+    } else {
+        numerator
+    };
+
+    let positive_divisor = if divisor < T::ZERO { -divisor } else { divisor };
```

Comment:
the inversion is wrong for one value in the integer range as an integer range is never symmetric
with n bits
[-2^(n - 1), 2^(n - 1)[
one value does not have a proper negative value and in fact folds onto itself when negated IIRC

```console
Python 3.10.12 (main, Jun 11 2023, 05:26:28) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy
>>> numpy.int64(-2**63)
-9223372036854775808
>>> a = numpy.int64(-2**63)
>>> -a
-9223372036854775808
>>>

Reply:
Though it might be ok in terms of bit pattern as unsigned, but I think we still need a unit test

Reply:
This is ok, I will add an explicit test for it but it's the expect behaviour

Reply:
yeah I went full circle on this one to see it was actually ok in the end